### DISCUSSION TITLE
runSQL error: duplicate key value violates unique constraint

### DISCUSSION THREAD

**Discussion by juniorforlife** [supabase/supabase#29074] [Questions]
I'm migrating my data from Firestore to Supabase and when I run `node json2supabase.js videos.json` it throws 
```
runSQL error: error: duplicate key value violates unique constraint "videos_pkey"
{
  ...
  code: '23505',
  detail: 'Key (id)=(US4BJjypi_Y) already exists.',
  ...
}
```
I checked my json file and there's no duplicate of `US4BJjypi_Y`.
Here's how the object looks like
```
{
    "id": "US4BJjypi_Y",
    "title": "Miền Đất Phúc - Tập 35 | HTV Phim Tình Cảm Việt Nam",
    "thumbnail": "https://i.ytimg.com/vi/US4BJjypi_Y/mqdefault.jpg"
}
```
And here's my table schema 
<img width="656" alt="Screenshot 2024-09-04 at 08 48 18" src="https://github.com/user-attachments/assets/b23c3f3e-d57c-45b2-b80a-0ef3715e46f8">
The reason I use type `text` is these are Youtube videos so I use their ids instead of Supabase auto-increment id.
Can anyone please show me what I'm doing wrong?

**Comment by GaryAustin1** [comment]
I'm not familiar with the conversion code you are running, but the Postgres error will not be wrong.  You are attempting to insert that key twice for some reason to the table.

**Reply by juniorforlife** [reply]
I'm just following the documentation https://supabase.com/docs/guides/resources/migrating-to-supabase/firestore-data#custom-hooks. First I export the Firestore collection to json then import it to Supabase
